xsd/parse.go

```diff
@@ -140,20 +176,33 @@ func Parse(docs ...[]byte) ([]Schema, error) {
 	for _, root := range schema {
 		tns := root.Attr("", "targetNamespace")
 		s := Schema{TargetNS: tns, Types: make(map[xml.Name]Type)}
+		sHash := hash(root)
```

Comment:
I'm a little concerned about this part. From my understanding of the `hash()` function, making changes to the tree structure in `root` could produce a different hash value. It looks like you don't make any transformations between here and when you lookup the schema on line 205, so there's no bug... yet :)

Reply:
Will think on this